pkg/mimo/actuator/manager.go

```diff
@@ -0,0 +1,182 @@
+package actuator
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/mimo/tasks"
+)
+
+const maxDequeueCount = 5
+
+type Actuator interface {
+	Process(context.Context) (bool, error)
+	AddTask(string, tasks.TaskFunc)
+	AddTasks(map[string]tasks.TaskFunc)
+}
+
+type actuator struct {
+	env env.Interface
+	log *logrus.Entry
+	now func() time.Time
+
+	clusterID string
+
+	oc  database.OpenShiftClusters
+	mmf database.MaintenanceManifests
+
+	tasks map[string]tasks.TaskFunc
+}
+
+func NewActuator(
+	ctx context.Context,
+	_env env.Interface,
+	log *logrus.Entry,
+	clusterID string,
+	oc database.OpenShiftClusters,
+	mmf database.MaintenanceManifests,
+	now func() time.Time) (Actuator, error) {
+	a := &actuator{
+		env:       _env,
+		log:       log,
+		clusterID: strings.ToLower(clusterID),
+		oc:        oc,
+		mmf:       mmf,
+		tasks:     make(map[string]tasks.TaskFunc),
+
+		now: now,
+	}
+
+	return a, nil
+}
+
+func (a *actuator) AddTask(u string, t tasks.TaskFunc) {
+	a.tasks[u] = t
+}
+
+func (a *actuator) AddTasks(tasks map[string]tasks.TaskFunc) {
+	maps.Copy(a.tasks, tasks)
+}
+
+func (a *actuator) Process(ctx context.Context) (bool, error) {
+	// Get the manifests for this cluster which need to be worked
+	i, err := a.mmf.GetByClusterID(ctx, a.clusterID, "")
+	if err != nil {
+		return false, err
```

Comment:
shall we log the error here? e.g., `a.log.WithError(....)` or `a.log.Error(....)` ?